tests/integration/targets/vyos_prefix_lists/tests/cli/merged.yaml

```diff
@@ -0,0 +1,76 @@
+---
+- debug:
+    msg: START vyos_prefix_lists merged integration tests on connection={{ ansible_connection }}
+
+- include_tasks: _remove_config.yaml
+
+- block:
+    - name: Merge the provided configuration with the existing running configuration
+      register: result
+      vyos.vyos.vyos_prefix_lists: &id001
+        config:
+          - afi: "ipv4"
+            prefix_lists:
+              - name: "AnsibleIPv4PrefixList"
+                description: "PL configured by ansible"
+                entries:
+                  - sequence: 2
+                    description: "Rule 2 given by ansible"
+                    action: "permit"
+                    prefix: "92.168.10.0/26"
+                    le: 32
+
+                  - sequence: 3
+                    description: "Rule 3"
+                    action: "deny"
+                    prefix: "72.168.2.0/24"
+                    ge: 26
+
+          - afi: "ipv6"
+            prefix_lists:
+              - name: "AllowIPv6Prefix"
+                description: "Configured by ansible for allowing IPv6 networks"
+                entries:
+                  - sequence: 5
+                    description: "Permit rule"
+                    action: "permit"
+                    prefix: "2001:db8:8000::/35"
+                    le: 37
+
+              - name: DenyIPv6Prefix
+                description: "Configured by ansible for disallowing IPv6 networks"
+                entries:
+                  - sequence: 8
+                    action: deny
+                    prefix: "2001:db8:2000::/35"
+                    le: 37
+        state: merged
+
+    - name: Assert that before dicts were correctly generated
+      assert:
+        that: "{{ result['before'] == [] }}"
+
+    - name: Assert that correct set of commands were generated
+      assert:
+        that:
+          - "{{ merged['commands'] | symmetric_difference(result['commands']) |length\
+            \ == 0 }}"
+
+    - name: Assert that after dicts were correctly generated
+      assert:
+        that:
+          - "{{ merged['after'][0] == result['after'][0] }}"
```

Comment:
This could be very much ineffective in case we have some unexpected entries, Better thing would be to compare whole list